### ISSUE TITLE
Path animation crashes on Android N

### ISSUE THREAD

**Issue by MizzleDK** [MizzleDK/IntroActivity#1]
As title. Not yet sure why it happens, but here's some of the crash log:

```
04-04 22:10:40.232 4640-4640/? A/DEBUG: *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
04-04 22:10:40.232 4640-4640/? A/DEBUG: Build fingerprint: 'google/angler/angler:N/NPC56W/2673096:user/release-keys'
04-04 22:10:40.233 4640-4640/? A/DEBUG: Revision: '0'
04-04 22:10:40.233 4640-4640/? A/DEBUG: ABI: 'arm64'
04-04 22:10:40.233 4640-4640/? A/DEBUG: pid: 4596, tid: 4634, name: RenderThread  >>> com.miz.introactivity <<<
04-04 22:10:40.233 4640-4640/? A/DEBUG: signal 6 (SIGABRT), code -6 (SI_TKILL), fault addr --------
04-04 22:10:40.242 4640-4640/? A/DEBUG: Abort message: 'Unsupported command: 3'
04-04 22:10:40.242 4640-4640/? A/DEBUG:     x0   0000000000000000  x1   000000000000121a  x2   0000000000000006  x3   0000000000000008
04-04 22:10:40.242 4640-4640/? A/DEBUG:     x4   00006e6174736e69  x5   0000000000000000  x6   000000792fcd1000  x7   0000000000000000
04-04 22:10:40.242 4640-4640/? A/DEBUG:     x8   0000000000000083  x9   ffffffffffffffdf  x10  0000000000000000  x11  0000000000000001
04-04 22:10:40.242 4640-4640/? A/DEBUG:     x12  0000000000000018  x13  0000000000000000  x14  0000000000000000  x15  0008e1786086642a
04-04 22:10:40.242 4640-4640/? A/DEBUG:     x16  000000792d339ec8  x17  000000792d2e37cc  x18  0000000000000000  x19  0000007911cf64f8
04-04 22:10:40.242 4640-4640/? A/DEBUG:     x20  0000000000000006  x21  0000007911cf6450  x22  0000000000000000  x23  00000079210ae610
04-04 22:10:40.242 4640-4640/? A/DEBUG:     x24  0000000000000000  x25  0000007911cf4990  x26  000000000000006d  x27  000000792c8d96a4
04-04 22:10:40.242 4640-4640/? A/DEBUG:     x28  00000000fffffff2  x29  0000007911cf4330  x30  000000792d2e0bf8
04-04 22:10:40.242 4640-4640/? A/DEBUG:     sp   0000007911cf4310  pc   000000792d2e37d4  pstate 0000000060000000
04-04 22:10:40.243 4645-4645/? I/art: JIT created with initial_capacity=64KB, max_capacity=64MB, compile_threshold=10000, save_profiling_info=1
04-04 22:10:40.251 4640-4640/? A/DEBUG:     #00 pc 000000000006a7d4  /system/lib64/libc.so (tgkill+8)
04-04 22:10:40.251 4640-4640/? A/DEBUG:     #01 pc 0000000000067bf4  /system/lib64/libc.so (pthread_kill+64)
04-04 22:10:40.251 4640-4640/? A/DEBUG:     #02 pc 000000000002386c  /system/lib64/libc.so (raise+24)
04-04 22:10:40.251 4640-4640/? A/DEBUG:     #03 pc 000000000001c2a4  /system/lib64/libc.so (abort+52)
04-04 22:10:40.251 4640-4640/? A/DEBUG:     #04 pc 000000000000ed74  /system/lib64/libcutils.so (__android_log_assert+224)
04-04 22:10:40.251 4640-4640/? A/DEBUG:     #05 pc 000000000003c15c  /system/lib64/libhwui.so
04-04 22:10:40.252 4640-4640/? A/DEBUG:     #06 pc 000000000003b71c  /system/lib64/libhwui.so (android::uirenderer::VectorDrawableUtils::verbsToPath(SkPath*, android::uirenderer::VectorDrawable::Path::Data const&)+132)
04-04 22:10:40.252 4640-4640/? A/DEBUG:     #07 pc 0000000000086258  /system/lib64/libhwui.so (android::uirenderer::VectorDrawable::FullPath::getUpdatedPath()+44)
04-04 22:10:40.252 4640-4640/? A/DEBUG:     #08 pc 0000000000085aa4  /system/lib64/libhwui.so (android::uirenderer::VectorDrawable::Path::draw(SkCanvas*, SkMatrix const&, float, float)+96)
04-04 22:10:40.252 4640-4640/? A/DEBUG:     #09 pc 00000000000868dc  /system/lib64/libhwui.so (android::uirenderer::VectorDrawable::Group::draw(SkCanvas*, SkMatrix const&, float, float)+204)
04-04 22:10:40.252 4640-4640/? A/DEBUG:     #10 pc 00000000000868dc  /system/lib64/libhwui.so (android::uirenderer::VectorDrawable::Group::draw(SkCanvas*, SkMatrix const&, float, float)+204)
04-04 22:10:40.252 4640-4640/? A/DEBUG:     #11 pc 0000000000086fac  /system/lib64/libhwui.so (android::uirenderer::VectorDrawable::Tree::getBitmapUpdateIfDirty()+144)
04-04 22:10:40.252 4640-4640/? A/DEBUG:     #12 pc 000000000004dcbc  /system/lib64/libhwui.so
04-04 22:10:40.252 4640-4640/? A/DEBUG:     #13 pc 0000000000045790  /system/lib64/libhwui.so
04-04 22:10:40.252 4640-4640/? A/DEBUG:     #14 pc 0000000000045644  /system/lib64/libhwui.so
04-04 22:10:40.252 4640-4640/? A/DEBUG:     #15 pc 00000000000603ec  /system/lib64/libhwui.so
04-04 22:10:40.252 4640-4640/? A/DEBUG:     #16 pc 00000000000310c4  /system/lib64/libhwui.so
04-04 22:10:40.252 4640-4640/? A/DEBUG:     #17 pc 000000000003275c  /system/lib64/libhwui.so
04-04 22:10:40.252 4640-4640/? A/DEBUG:     #18 pc 0000000000037cc4  /system/lib64/libhwui.so (android::uirenderer::renderthread::RenderThread::threadLoop()+152)
04-04 22:10:40.252 4640-4640/? A/DEBUG:     #19 pc 00000000000151f8  /system/lib64/libutils.so (android::Thread::_threadLoop(void*)+272)
04-04 22:10:40.252 4640-4640/? A/DEBUG:     #20 pc 000000000009b474  /system/lib64/libandroid_runtime.so (android::AndroidRuntime::javaThreadShell(void*)+116)
04-04 22:10:40.252 4640-4640/? A/DEBUG:     #21 pc 00000000000673dc  /system/lib64/libc.so (__pthread_start(void*)+208)
04-04 22:10:40.252 4640-4640/? A/DEBUG:     #22 pc 000000000001d510  /system/lib64/libc.so (__start_thread+16)
```

**Comment by MizzleDK** [comment]
It seems like loading of vector files was changed in N, making it more efficient. This is probably a bug as a result of that.

**Comment by nyelito** [comment]
Hey man, I had this same bug in my app and now getting it when trying to integrate IntroActivity. Fortunately I found the solution.

Here is the relevant part of the stack trace I get
`Caused by: android.content.res.Resources$NotFoundException: Drawable com.nyelito.dactyl:drawable/done_to_next with resource ID #0x7f020080
                                                                     Caused by: java.lang.IllegalArgumentException: Invalid pathData. Failure occurred at position 0 of path: 33mM 18.0,32.34 l 0.0,0.0 L 18.0,32.34 H 18.0 v 0.0 h 0.0 L 9.66,24.0 l -2.83,2.83 L 18.0,38.0 l 24.0,-24.0 l -2.83,-2.83 L 18.0,32.34
                                                                        at android.util.PathParser.nCreatePathDataFromString(Native Method)
                                                                        at android.util.PathParser.-wrap1(PathParser.java)`

The solution is given by an Android engineer here: https://code.google.com/p/android/issues/detail?id=206018

Basically, you need to remove "33m" from the beginning of the path, and all paths that start with something like that, and it should work fine.

**Comment by MizzleDK** [comment]
Hi! Yeah, I filed the bug you linked to :smiley: For some reason I never added a commit with the fix, though. Feel free to create a PR, if you want!

**Comment by nyelito** [comment]
Haha awesome, I put in two separate PR's, one for each file, because I just made the fixes through Github's edit function.  :P

**Comment by MizzleDK** [comment]
Thanks, I'll take a look at it as soon as possible :smile:

**Comment by ghost** [comment]
This still opens, any solution?
